lib/vrl/stdlib/src/parse_query_string.rs

```diff
@@ -0,0 +1,131 @@
+use bytes::Buf;
+use std::collections::btree_map::Entry;
+use std::collections::BTreeMap;
+use url::form_urlencoded;
+use vrl::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ParseQueryString;
+
+impl Function for ParseQueryString {
+    fn identifier(&self) -> &'static str {
+        "parse_query_string"
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[Example {
+            title: "parse query string",
+            source: r#"parse_query_string("foo=1&bar=2")"#,
+            result: Ok(r#"
+                {
+                    "foo": "1",
+                    "bar": "2"
+                }
+            "#),
+        }]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Compiled {
+        let value = arguments.required("value");
+        Ok(Box::new(ParseQueryStringFn { value }))
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            kind: kind::BYTES,
+            required: true,
+        }]
+    }
+}
+
+#[derive(Debug, Clone)]
+struct ParseQueryStringFn {
+    value: Box<dyn Expression>,
+}
+
+impl Expression for ParseQueryStringFn {
+    fn resolve(&self, ctx: &mut Context) -> Resolved {
+        let query_string = self.value.resolve(ctx)?.try_bytes()?;
+        let mut result: BTreeMap<String, Value> = BTreeMap::new();
+        let parsed = form_urlencoded::parse(query_string.bytes());
+        for (k, v) in parsed {
+            let value = v.as_ref().into();
+            let entry = result.entry(k.as_ref().to_owned());
```

Comment:
```suggestion
            let entry = result.entry(k.into_owned());
```

`Cow` has a [`into_owned`](http://doc.rust-lang.org/1.50.0/alloc/borrow/enum.Cow.html#method.into_owned), which does the same.

Reply:
Thanks, updated.